iota/filters.py

```diff
@@ -142,3 +143,67 @@ def _apply(self, value):
           'result_type': self.result_type.__name__,
         },
       )
+
+
+class AddressNoChecksum(Trytes):
+  """
+  Validates a sequence as a sequence of trytes.
+  """
+
+  def _apply(self, value):
+    # noinspection PyTypeChecker
+    value =\
+      self._filter(
+        filter_chain = f.Type((binary_type, bytearray, text_type, TryteString)),
+        value = value,
+      ) # type: TrytesCompatible
+
+    if self._has_errors:
+      return None
+
+    # If the incoming value already is an Address then we just make sure it has no checksum
+    if isinstance(value, Address):
+      if value.checksum and not value.is_checksum_valid():
+        return self._invalid_value(
+        value     = value,
+        reason    = self.ADDRESS_BAD_CHECKSUM,
+        exc_info  = True,
+
+        template_vars = {
+          'result_type': self.ADDRESS_BAD_CHECKSUM,
+        },
+      )
+      return Address(value.address)
+
+    # First convert to a generic TryteString, to make sure that the
+    # sequence doesn't contain any invalid characters.
+    try:
+      value = TryteString(value)
+    except ValueError:
+      return self._invalid_value(value, self.CODE_NOT_TRYTES, exc_info=True)
+
+    # Now coerce to the expected type and verify that there are no
+    # type-specific errors.
+    try:
+      addy = Address(value)
+      if addy.checksum and not addy.is_checksum_valid():
+        return self._invalid_value(
+        value     = addy,
+        reason    = self.ADDRESS_BAD_CHECKSUM,
+        exc_info  = True,
+
+        template_vars = {
+          'result_type': self.ADDRESS_BAD_CHECKSUM,
+        },
+      )
+      return Address(addy.address)
+    except ValueError:
+      return self._invalid_value(
+        value     = value,
+        reason    = self.CODE_WRONG_FORMAT,
+        exc_info  = True,
+
+        template_vars = {
+          'result_type': self.result_type.__name__,
+        },
+      )
```

Comment:
☂️ Is it possible to re-engineer this method so that the checksum checker code only appears once?